src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -14,9 +14,11 @@
 
     @Override
     public User login(String login, String password) throws AuthenticationException {
-        Optional<User> optionalUser = userService.getByLogin(login);
-        if (optionalUser.isPresent() && optionalUser.get().getPassword().equals(password)) {
```

Comment:
we could call a private method in the right part of the statement in order not to throw the exception twice

Reply:
changed